### ISSUE TITLE
Dedicated `change_metadata` table

### ISSUE THREAD

**Issue by samuelstroschein** [opral/lix-sdk#87]
### Context

* change metadata makes sense (for apps, not plugins!)
* because metadata is dynamic in size, it should be its own table

@samuel.stroschein said in [LIXDK-169](https://linear.app/opral/issue/LIXDK-169/add-confirm-changes-action-via-change-sets#comment-8d908879):

> @martin.lysk1 yep you're right. the `change.metadata` field makes sense because APPS can add metadata to changes. I can implement change sets and tags via the metadata field without requiring lix to implement them 🤯 That entails though:
>
> * plugins DON'T report metadata (as they do right now), only snapshots
> * we should likely implement Metadata as own table due to the same insight as [LIXDK-166](https://linear.app/opral/issue/LIXDK-166/investigate-if-dedicated-snapshot-table-is-benefitial))

### Proposal 

1. metadata table.

```diff
table change {
   ...
-  meta: JSON
+  metadata_id: TEXT
}
```

```
table metadata {
  id: UUID_V7
  value: JSON
}
```

2. Discourage/disallow plugins to set metadata (they report a snapshot)

cc @martin.lysk1 agree?

**Comment by martin-lysk** [comment]
Just to be sure - Can you describe the difference between apps and plugins?

**Comment by samuelstroschein** [comment]
**Plugins**

Plugin detect changes and conflicts in file formats. 

**Apps**

Provide a UI and functionality on top lix change control. 

**Difference**

Plugins instruct lix on how to deal with file formats. 

Good enough? We should add the definitions to the docs then

**Comment by martin-lysk** [comment]
Ok - in my own words: <br>A plugin is provided with the before and after states of a file and extracts all information that represents a snapshot of the entities within that file.<br><br>In case of a \*.md file that could be the whole file (because we have no identifier but the file name).<br><br>In case of inlang the plugin compares two states of a sqlite file, extracts changed rows (each having a ID) and stores them as a snapshot. 

\-> Makes sense

What i don't understand is the role app's play here - <br>A Plugin does **NOT(?)** enrich the change information with metadata. What, why and when would apps store metadata than?

**Comment by samuelstroschein** [comment]
Good explanation although the explanation is specific to `plugin.detectChanges`. The generic description for a plugin would go along the lines of "A plugin provides lix with change information for a file format". 

> What i don't understand is the role app's play here - <br>​A Plugin does **NOT(?)** enrich the change information with metadata. What, why and when would apps store metadata than?

Easier to explain on a call. 

TL;DR apps can enrich changes that have been reported by plugins however they need. 

For example, the CSV app can implement change tags via `change.metadata` without support from the lix SDK as a workaround/test case.

**Comment by samuelstroschein** [comment]
Cancelled. 

* needs more research on how to merge [https://discord.com/channels/897438559458430986/1295284074445013036/1295448160784875563](https://discord.com/channels/897438559458430986/1295284074445013036/1295448160784875563) 
* csv demo app can hack around metadata by storing metadata in own database
